packages/bot-web-ui/src/pages/dashboard/announcements/announcements.tsx

```diff
@@ -0,0 +1,173 @@
+    const [stored_notifications, setStoredNotifications] = React.useState({} as Record<string, boolean>);
+    const history = useHistory();
+    const [notifications, setNotifications] = useState([] as TNotifications[]);
+
+    const storeDataInLocalStorage = (temp_data: Record<string, boolean>) => {
+        localStorage?.setItem('bot-announcements', JSON.stringify(temp_data));
+    };
+
+    const modalButtonAction = (announce_id: string, announcement: TAnnouncement) => () => {
+        setSelectedAnnouncement(announcement);
+        setIsAnnounceDialogOpen(true);
+        setIsOpenAnnounceList(prev => !prev);
+
+        let data: Record<string, boolean> | null = null;
+        data = JSON.parse(localStorage.getItem('bot-announcements') ?? '{}');
+
+        storeDataInLocalStorage({ ...data, [announce_id]: false });
+        const temp_notifications = updateNotifications();
+        setStoredNotifications(temp_notifications);
+    };
+
+    const handleRedirect = (url: string) => {
+        if (history) {
+            history.push(url);
+        }
+    };
+
+    const updateNotifications = () => {
+        let data: Record<string, boolean> | null = null;
+        data = JSON.parse(localStorage.getItem('bot-announcements') ?? '{}');
+        const tmp_notifications: TNotifications[] = [];
+        const temp_localstorage_data: Record<string, boolean> | null = {};
+        BOT_ANNOUNCEMENTS_LIST.map(item => {
+            let is_not_read = true;
+            if (data && Object.hasOwn(data, item.id)) {
+                is_not_read = data[item.id];
+            }
+            tmp_notifications.push({
+                key: item.id,
+                icon: <item.icon announce={is_not_read} />,
+                title: <TitleAnnounce title={item.title} announce={is_not_read} />,
+                message: <MessageAnnounce message={item.message} date={item.date} announce={is_not_read} />,
+                buttonAction: getButtonAction(item, modalButtonAction, handleRedirect),
+                actionText: item.actionText,
+            });
+            temp_localstorage_data[item.id] = is_not_read;
+        });
+        setNotifications(tmp_notifications);
+        return temp_localstorage_data;
+    };
+
+    useEffect(() => {
+        const temp_localstorage_data = updateNotifications();
+        storeDataInLocalStorage(temp_localstorage_data);
+        setStoredNotifications(temp_localstorage_data);
+    }, []);
+
+    const countActiveAnnouncements = (): number => {
+        return Object.values(stored_notifications as Record<string, boolean>).reduce(
+            (count: number, value: boolean) => {
+                return value === true ? count + 1 : count;
+            },
+            0
+        );
+    };
+
+    useEffect(() => {
+        countActiveAnnouncements();
+    }, [stored_notifications]);
+
+    const number_ammount_announce = countActiveAnnouncements();
```

Comment:
@maryia-matskevich-deriv 
why are we calling countActiveAnnouncements() like this ?. 
Wouldn't it compute again when `stored_notifications` changes ?
Also can we use a separate use state to keep track of `number_ammount_announce` ?

Reply:
Also can we change the name to number_of_announcement ?

Reply:
@rupato-deriv 
We need to call the method **countActiveAnnouncements()** whenever **stored_notifications** changes, that's why we use this useEffect with the dependency. 
_Example of a situation:_ 
1) The user opens the list with announcements and clicks on 1 of them
2) =>  it becomes inactive
3) we need to calculate again how many active announcements we have and show, so we use the method `countActiveAnnouncements()`

About this request: "_Also can we use a separate use state to keep track of number_ammount_announce_ ?"
Once `stored_notifications` updates the component renders as we expect, we don't need any other reason to render the component again. I don't think it's a good idea to over-render. 🫤 You can check it

Reply:
let's wrap it with useMemo then @maryia-matskevich-deriv and put the `stored_notifications` as dependency 

I think it will work the same with more proficiency. Currently it will be called with each rerender.

Reply:
@sandeep-deriv I think we don't need it here **useMemo** even, look at the video recording, it renders 1 time when we click on the announcement item... we have **not** any extra renders at all 🫤
https://github.com/user-attachments/assets/8a8526b2-e0ed-4538-b470-5bae438445c0